src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -288,19 +288,10 @@ public void WaitForCancelCompletion()
                 {
                     taskCleanedUp = _requestTask.Wait(BuildParameters.RequestBuilderShutdownTimeout);
                 }
-                catch (AggregateException e)
+                catch (AggregateException e) when (e.Flatten().InnerExceptions.All(ex => ex is TaskCanceledException || ex is OperationCanceledException))
```

Comment:
Add comment:
Verifies that all the inner exceptions are cancellation-related exceptions.